Swap.jl

```diff
@@ -0,0 +1,66 @@
+include("Solution.jl")
+
+DEBUG = true
+
+struct Swap
+    data::TSP
+    solution::Solution
+ end
+ 
+function nodeCost(route::Vector{Int}, index::Int64, weights::Matrix{Float64})
+    size = length(route)
+    value = route[index]
+
+    if index == 1 
+        prev = route[size]
+    else
+        prev = route[index - 1]
+    end
+
+    if index == size
+        next = route[1]
+    else
+        next = route[index + 1]
+    end
+    return weights[prev, value] + weights[value, next]
+
+end
+
+ function eval(swap::Swap, pos1::Int64, pos2::Int64)::Int64
```

Comment:
I agree that doing the swap in order to evaluate keeps things cleaner, and although the `eval` subroutine must be as efficient as possible (it's inside two loops), the two `move!` calls should hardly impact on the overall runtime.

It should be noted that the case  `pos1 + 1 = pos2` yields a simpler calculation, but I think there is no need to implement this condition.

Also, since weights is being passed as a Float64 matrix, the function should return the cost of the operation as a Float64.